pkg/snmp/traps/formatter.go

```diff
@@ -340,12 +340,17 @@ func formatType(variable gosnmp.SnmpPDU) string {
 }
 
 func formatValue(variable gosnmp.SnmpPDU) interface{} {
-	switch variable.Value.(type) {
-	case []byte:
-		return string(variable.Value.([]byte))
-	default:
-		return variable.Value
+	value := variable.Value
+
+	if bytes, ok := value.([]byte); ok {
+		value = string(bytes)
+	}
+
+	if variable.Type == gosnmp.ObjectIdentifier {
+		return NormalizeOID(value.(string))
 	}
+
+	return value
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Could you add a unit test for the new change?‏